department-of-veterans-affairs/vets-website
# [MHV-57335] Your account is no longer connected to TG issue when in a new tab

> ## Summary
> 
> - When a current, logged-in user refreshed or switched to a new tab their screen on a message details page, a blocked TG alert appeared incorrectly, although the user should still be connected to that TG.
> - This happened because the list of recipients, necessary to discern if the user is blocked from the TG, was not retrieved initially by the allRecipients api call during the refresh or switch. 
> - To fix this, the allRecipients api call should be made first when the page refreshes or a new tab is opened at the top layer of the application._
> 
> ## Related issue(s)
> 
> ### [[MHV-57335]](https://jira.devops.va.gov/browse/MHV-57335) "Your account is no longer connected to TG" issue when users open a message in a new tab
> 
> #### Description
>  "Your account is no longer connected to TG" when message is opened in a new tab
> 
> Scenario: Error Message Displayed When Opening Messages in New Tabs
> Given A user is logged into the [VA.gov](http://va.gov/) GUI
> And Navigates to the SM home page
> And The user opens a message in a new tab
> When The user Switches to the newly opened tab.
> Then an error message indicating that their account is no longer connected to TG is returned to the user.
>  
> #### SLACK conversation: 
> https://dsva.slack.com/archives/C03T8JY2DM2/p1713791758437029
> 
> ## Testing done
> 
> - _Manual testing was performed_
> -  To test, navigate to any message that is not older than 45 days. 
> - The TG alert should NOT appear.
> 
> ## Screenshots
> 
> _Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._
> 
> |         | Before | After |
> | ------- | ------ | ----- |
> | Mobile  |        |       |
> | Desktop |        |       |
> 
> ## What areas of the site does it impact?
> 
> *Va.gov - Secure Messaging*
> 
> ## Acceptance criteria
> 
> ### Quality Assurance & Testing
> 
> - [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
> - [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
> - [ ] Linting warnings have been addressed
> - [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
> - [ ] Screenshot of the developed feature is added
> - [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed
> 
> ### Error Handling
> 
> - [ ] Browser console contains no warnings or errors.
> - [ ] Events are being sent to the appropriate logging solution
> - [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)
> 
> ### Authentication
> 
> - [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user
> 
> ### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:
> 
> - [ ] The vets-website header does not contain any web-components
> - [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
> - [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions
> 
> ## Requested Feedback
> 
> (OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_
>
